### PR TITLE
[Analyzer] Property definition default value

### PR DESCRIPTION
**Pull request by deepvyas**
Hey!

Type: new feature| documentation

Link to issue: https://github.com/ovr/phpsa/issues/166

This pull request affects the following components **(please check boxes):**

* [ ] Core
* [x] Analyzer
* [ ] Compiler
* [ ] Control Flow Graph
* [x] Documentation

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/ovr/phpsa/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: This adds an analyzer which use of non-required default value for property. (PropertyProperty is not being registered)

Thanks

### CODE REVIEW

File: tests/analyze-fixtures/Statement/PropertyDefinitionDefaultValue.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+namespace Tests\Analyze\Fixtures\Statement;
+
+class PropertyDefinitionDefaultValue
+{
+    public $a = null;
+}
+
+class PropetyDefinitionNoDefaultValue
+{
+    public $a;
```
Comment (ddmler) [inline tests/analyze-fixtures/Statement/PropertyDefinitionDefaultValue.php]: you can just put this into the class before. No need for 2 classes :)

**Comment by ddmler** [comment]
I added support for PropertyProperty in master. Rebase your branch and we'll see if it's working.

**Comment by deepvyas** [comment]
I am a bit new to git rebase and hence and hence not doing it now. Will do it once I understand how it works.

**Comment by deepvyas** [comment]
Post Rebase. The error still remains its the same one i mentioned in the issue.
I am unable to figure out what this points to.

**Comment by ddmler** [comment]
It's now line 7 instead of 6 we fixed the line number.

**Comment by ddmler** [comment]
@deepvyas 
So I checked it, if you add to your analyzer:
`use PHPSA\CompiledExpression;`
and change in your test the 2nd variable name (you can't name both a)
it works.

Unfortunately we have some conflicts on the branch do you know how to resolve them with rebase?

**Comment by deepvyas** [comment]
@ddmler 
I have made the necessary changes and build passes as well.
Sorry for the trouble. It took me some time.
Cheers!!

**Comment by ddmler** [comment]
No problem @deepvyas Thanks for the good work :+1:

**Pull request merged**